src/iwconnect.c

```diff
@@ -48,8 +49,9 @@ int exec_iwconnect(const char *ssid, const char *password)
 		os_strncpy((char*)&sta_conf.password, password, 32);
   }
 
-	wifi_station_set_config(&sta_conf);		
 	wifi_station_disconnect();
+	wifi_station_set_config(&sta_conf);
+	wifi_station_set_hostname(env_get("hostname"));
```

Comment:
What if hostname is not set? Is wifi_station_set_hostname() okay will NULL argument?

Reply:
Yes, it seems so. If I replace it with "wifi_station_set_hostname(NULL);", frankenstein connects using the old ESP_xxxx as name.

Reply:
Okay then, merging. Thanks.